hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieHadoopFsRelationFactory.scala

```diff
@@ -103,6 +103,21 @@ abstract class HoodieBaseHadoopFsRelationFactory(val sqlContext: SQLContext,
       // For older tables the partition type may not be available so falling back to partition fields in those cases
       tableConfig.getPartitionFields.orElse(Array.empty).toSeq
     }
+  } else {
+    Seq.empty
+  }
+
+  protected lazy val partitionColumnsToRead: Seq[String] = (
+    shouldExtractPartitionValuesFromPartitionPath,
+    keygenTypeHasVariablePartitionCols,
+    partitionColumnsHasPrecombine
+  ) match {
+    case (true, _, _) => Seq.empty
+    case (false, true, false) => variableTimestampKeygenPartitionCols
+    case (false, true, true) if variableTimestampKeygenPartitionCols.contains(preCombineFieldOpt.get) => variableTimestampKeygenPartitionCols
+    case (false, true, true) => variableTimestampKeygenPartitionCols :+ preCombineFieldOpt.get
```

Comment:
Could we rewrite this to be legible, sth like
```
case (false, true, true) => if (variableTimestampKeygenPartitionCols.contains(preCombineFieldOpt.get)) { 
  variableTimestampKeygenPartitionCols
} else {
  variableTimestampKeygenPartitionCols :+ preCombineFieldOpt.get
}
```

Reply:
😂